ci/version.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: Apache-2.0
+
+import subprocess
+import sys
+import os
+
+
+def set_version(current_version, new_version, dry_run=False):
+    """Tries to update the version from current_version to new_version."""
+    print(f"Modifying version from {current_version} to {new_version}...")
+    if dry_run:
+        print("  (dry run, won't actually make the change)")
+
+    # Read the template.
+    with open("version-diff-template", "r") as f:
+        template = f.read()
+
+    # Fill out the template.
+    patch = template.format(frm=current_version, to=new_version).encode("utf-8")
+
+    # Try to apply the patch.
+    output = subprocess.run(["git", "apply", "--check"], cwd="..", input=patch)
+    if output.returncode:
+        print("Patch is invalid. Something is out of sync: check the version")
+        print("file and try regenerating the patch template.")
+        sys.exit(1)
+    print("Generated patch seems valid.")
+
+    # Stop here if this is a dry run.
+    if dry_run:
+        return
+
+    # Apply the patch.
+    subprocess.run(["git", "apply"], cwd="..", input=patch, check=True)
+
+    # Update the current version file.
+    with open("version", "w") as f:
+        f.write(new_version)
+
+    print("Version modified.")
+
+
+def escape(s):
+    """Escapes { and } sequences in a diff template file."""
+    return s.replace("{", "{{").replace("}", "}}")
+
+
+if __name__ == "__main__":
+
+    # The rest of the script assumes that we're running from the script
+    # directory, so make sure that's where we are.
+    os.chdir(os.path.dirname(os.path.realpath(__file__)))
+
+    # Read the current version.
+    with open("version", "r") as f:
+        current_version = f.read().strip()
+
+    if len(sys.argv) == 2 and sys.argv[1] == "get":
```

Comment:
Please use [`argparse`](https://docs.python.org/3/library/argparse.html) for all of this argument handling. It handles subcommands and will make the handling much easier to read.